cloudwatch/metrics.go

```diff
@@ -12,39 +12,41 @@ import (
 )
 
 type Metric []string
+type MetricsRequest map[string]interface{}
 
 // GetMetricWidget gets a metric widget image for an instance id
-func (c *Cloudwatch) GetMetricWidget(ctx context.Context, metrics []Metric, period int64, start, end string) ([]byte, error) {
-	if len(metrics) == 0 || period == 0 || start == "" || end == "" {
+// https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/CloudWatch-Metric-Widget-Structure.html
+// https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/viewing_metrics_with_cloudwatch.html
+// https://docs.aws.amazon.com/AmazonECS/latest/developerguide/cloudwatch-metrics.html
+//
+// Example metrics request
+// {
+//   "metrics": [
+//     [ "AWS/ECS", "CPUUtilization", "ClusterName", "spinup-000393", "ServiceName", "spinup-0010a3-testsvc" ]
+//   ],
+//   "stat": "Average"
+//   "period": 300,
+//   "start": "-P1D",
+//   "end": "PT0H"
+// }
+func (c *Cloudwatch) GetMetricWidget(ctx context.Context, req MetricsRequest) ([]byte, error) {
```

Comment:
much cleaner with this `MetricsRequest` 💯

Reply:
🦖